lib/upgrade/etcd/conversionv1v3/workloadendpoint_test.go

```diff
@@ -0,0 +1,393 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conversionv1v3
+
+import (
+	"testing"
+
+	cnet "net"
+	. "github.com/onsi/gomega"
+
+	apiv1 "github.com/projectcalico/libcalico-go/lib/apis/v1"
+	"github.com/projectcalico/libcalico-go/lib/apis/v1/unversioned"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/net"
+	"github.com/projectcalico/libcalico-go/lib/numorstring"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var wepTable = []struct {
+	description string
+	v1API       unversioned.Resource
+	v1KVP       *model.KVPair
+	v3API       apiv3.WorkloadEndpoint
+}{
+	{
+		description: "fully populated WEP",
+		v1API: apiv1.WorkloadEndpoint{
+			Metadata: apiv1.WorkloadEndpointMetadata{
+				Name:             "eth0",
+				Workload:         "default.frontend-5gs43",
+				Orchestrator:     "k8s",
+				Node:             "TestNode",
+				ActiveInstanceID: "1337495556942031415926535",
+				Labels:           makeLabelsV1(),
```

Comment:
I recommend adding a test without any labels.

A "minimalistic" test, if you will - only the bare minimum, just in case such a wep exists.